tests/integration/testlib/ovslib.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+import copy
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import OVSBridge
+from libnmstate.schema import OVSBridgePortType
+
+
+class Bridge:
+    def __init__(self, name):
+        self._name = name
+        self._ifaces = [
+            {
+                Interface.NAME: name,
+                Interface.TYPE: InterfaceType.OVS_BRIDGE,
+                OVSBridge.CONFIG_SUBTREE: {},
+            }
+        ]
+        self._bridge_iface = self._ifaces[0]
+
+    def set_options(self, options):
+        self._bridge_iface[OVSBridge.CONFIG_SUBTREE][
+            OVSBridge.OPTIONS_SUBTREE
+        ] = options
+
+    def add_system_port(self, name):
+        self._add_port(name, OVSBridgePortType.SYSTEM)
+
+    def add_internal_port(self, name, ipv4_state):
+        self._add_port(name, OVSBridgePortType.INTERNAL)
+        self._ifaces.append(
+            {
+                Interface.NAME: name,
+                Interface.TYPE: InterfaceType.OVS_INTERFACE,
+                Interface.IPV4: ipv4_state,
+            }
+        )
+
+    def _add_port(self, name, _type):
+        self._bridge_iface[OVSBridge.CONFIG_SUBTREE].setdefault(
+            OVSBridge.PORT_SUBTREE, []
+        ).append({OVSBridge.PORT_NAME: name, OVSBridge.PORT_TYPE: _type})
+
+    @contextmanager
+    def created(self):
```

Comment:
Using verbs in a past tense is strange.
Why not `create`?

Another option would be to use `__enter__` and `__exit__` directly.
This will yield:
```
bond = Bond (..)
bond....
with bond as desired_state:
    ...
```
(but I leave it up to you, it is just a suggestion)

Reply:
>  Why not create?
Because in past tense it makes sense as a context "with created bridge, I want to do X and Y". But since Python uses it with present, I will change it.

Reply:
I never heard about a past sense unless it is a question. A context is usually an object, therefore the name should be a noun but in some cases a verb is also file (e.g. ‘open’).
‘enter/exit’ are separated from the class instantiation and fit well with the later creation.

Reply:
Sure, just wanted it to make sense when read. Unrelated to programming. Will change.

Ahaaa, sorry, got a little rusty with Python. Anyway this looks weird, that you treat a value (without `()`) as something that returns context.

```python
with bond as desired_state:
   ...
```

Also, I like to use `@contextmanager` instead of magical methods.

Reply:
I do not mind you using it, but that decorator is the magic not the actual implementation of a python context-manager. The decorator is creating a class that have these “magic” methods.
Anyway, it is not worth debating this, do what makes you comfortable with.

Reply:
Was refering to `__magic_methods__` in the code. Petr doesn't like how they look. That's all. Debate debate!

Reply:
Done :fireworks: